### ISSUE TITLE
Can't finish the workshop

### ISSUE THREAD

**Issue by marttinn** [asbjornenge/thinking-in-react#19]
Once I got to the last excercise. (And required React-Dom) something went wrong and I couldn't continue because thinking-in-react verify/run gave me this: 

`martinn@The-Macbook-Air:~/Developer/nodeschoooling/think-react% thinking-in-react verify ex4.js                                                    16:43  martinn@The-Macbook-Air
module.js:328
    throw err;
    ^

Error: Cannot find module '/Users/martinn/Developer/nodeschoooling/think-react/node_modules/react/addons'
    at Function.Module._resolveFilename (module.js:326:15)
    at Function.Module._load (module.js:277:25)
    at Module.require (module.js:354:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/usr/local/lib/node_modules/thinking-in-react/exercises/spec_5.js:1:75)
    at Module._compile (module.js:410:26)
    at Module._extensions..js (module.js:417:10)
    at Object.require.extensions.(anonymous function) [as .js] (/usr/local/lib/node_modules/thinking-in-react/node_modules/babel-core/lib/api/register/node.js:214:7)
    at Module.load (module.js:344:32)
    at Function.Module._load (module.js:301:12)`

**Comment by asbjornenge** [comment]
@marttinn hi! Thanks for reporting this :+1: I wonder if this is caused by the new React v15. Could you give it a quick try with 14? If it works, I'll get started on a fix :wink:

**Comment by cbetancourt** [comment]
The `run` and `verify` steps fail when executed with v15.1.0, but the `server` bundles and deploys correctly. I have successfully executed all steps with `react@0.14.8` and `react-dom@0.14.8`.

**Comment by asbjornenge** [comment]
@cbetancourt thanks a bunch! I'll get on a fix for that :smile:

**Comment by asbjornenge** [comment]
Fixed and published @ v3.1.0 :grin: :tada: :rocket: 

:heart: if you can verify :stuck_out_tongue: 

cc @cbetancourt @marttinn

**Comment by cbetancourt** [comment]
@asbjornenge: I upgraded to v3.5.1 and while we get further than before, the Inverse dataflow test spec fails with the following:
```
  1) Inverse dataflow should update state on FilterableProductTable when modifying inputs:
     Uncaught AssertionError: false == true
      at /Users/cbetancourt/.npm-packages/lib/node_modules/thinking-in-react/exercises/spec_5.js:22:17
      at nextTickCallbackWith0Args (node.js:420:9)
      at process._tickCallback (node.js:349:13)
```

Even your [published solution](https://github.com/asbjornenge/thinking-in-react/blob/master/exercises/step_5__add_inverse_data_flow/solution/solution.js) fails this test with versions 0.14.8 and 15.1.0 of `react` and `react-dom`.

Let me know if you need me to test a new release. Thank you.

**Comment by asbjornenge** [comment]
@cbetancourt thanks for helping to test this :smile: I've published a new version `v3.1.1` which should fix the problem (I hope :stuck_out_tongue:) However, the problem stems from an issue with `react-addons-test-utils` in that it needs to be requiring the same `react` as the solution. In previous versions of `react` I could work around this by using the bundled `react/addon`, but as of  v15 that is removed. So I've updated instructions in `step2` to include installing `react-addons-test-utils`. With that, it should work! *fingers crossed* :stuck_out_tongue_closed_eyes:
